app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,40 @@
+class PostsController < ApplicationController
+    def create
+        @post = current_user.posts.build(post_params)
+        if @post.save
+            redirect_back(fallback_location: root_path)
+            flash[:notice] = "Post successfully created"
+        else
+            flash[:error] = @post.errors.full_messages
+            redirect_back(fallback_location: root_path)
+        end
+    end
+
+    def destroy
+        @post = Post.find(params[:id])
```

Comment:
we are repeating `@post = Post.find(params[:id])` for edit, update and destroy, that would be a good piece of code to put in a private method and set a before_action to make available to those methods.